x/ccv/provider/types/proposal.go

```diff
@@ -76,6 +92,32 @@ func (cccp *ConsumerAdditionProposal) ValidateBasic() error {
 	if cccp.SpawnTime.IsZero() {
 		return sdkerrors.Wrap(ErrInvalidConsumerAdditionProposal, "spawn time cannot be zero")
 	}
+
+	if _, err := sdktypes.NewDecFromStr(cccp.ConsumerRedistributionFraction); err != nil {
+		return sdkerrors.Wrapf(ErrInvalidConsumerAdditionProposal, "consumer redistribution fraction is invalid: %s", err)
+	}
+
+	if cccp.BlocksPerDistributionTransmission < 1 {
+		return sdkerrors.Wrap(ErrInvalidConsumerAdditionProposal, "blocks per distribution transmission cannot be < 1")
+	}
+
+	if cccp.HistoricalEntries < 1 {
+		return sdkerrors.Wrap(ErrInvalidConsumerAdditionProposal, "historical entries cannot be < 1")
+	}
+
+	zeroDuration := time.Duration(0)
```

Comment:
Just a hunch. Is there a reason why they should be allowed to be 0?

`CcvTimeoutPeriod == 0` would timeout IBC packets immediately after send/receive.
`TransferTimeoutPeriod == 0` would timeout IBC transfer packets in after send/receive
`UnbondingPeriod == 0` means that tokens would be liquid as soon as they are undelegated

I may be reading into this wrong. Please correct me if I am mistaken

Reply:
`UnbondingPeriod == 0` I think this might break ibc in funky ways. Ibc uses it when resolving trusting periods and they have some weird rules. Ie.g trustingPeriod < unbondingPeriod which means you wouldn't be able to use ibc.